src/Altinn.App.Core/Internal/Process/ProcessReader.cs

```
@@ -168,4 +169,22 @@ public List<ProcessElement> GetAllFlowElements()
         flowElements.AddRange(GetEndEvents());
         return flowElements;
     }
+
+    /// <summary>
+    /// Get the AltinnTaskExtension for a given task
+    /// </summary>
+    /// <param name="elementId"></param>
+    /// <returns></returns>
+    /// <exception cref="ProcessException"></exception>
+    public AltinnTaskExtension? GetAltinnTaskExtension(string elementId)
+    {
+        ProcessElement? flowElement = GetFlowElement(elementId);
+
+        if (flowElement is ProcessTask processTask)
+        {
+            return processTask.ExtensionElements?.TaskExtension ?? throw new ProcessException("No AltinnTaskExtension found on task");
+        }
+
+        return null;
```

Kan du forklare når denne skal returnere `null` og når den skal kaste `ProcessException`

---

Et `ProcessElement` er ikke nødvendigvis en `ProcessTask` kan f.eks. være en `StartEvent`. Hvis det er en `StartEvent` så har den ingen TaskExtension og det er gyldig. Derimot hvis det er en `ProcessTask` så skal den ha en TaskExtension ellers er det en ugyldig prosess definisjon (fail fast)

---

Da bør det vel kanskje stå i en kommentar?